packages/google_maps_flutter/example/test_driver/google_maps.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_driver/driver_extension.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+
+import 'google_maps_test_controller.dart';
+
+const CameraPosition _kInitialCameraPosition =
+    CameraPosition(target: LatLng(0, 0));
+
+class GoogleMapTest extends StatefulWidget {
+  GoogleMapTest(
+    this.mapState,
+    this._controllerCompleter,
+  );
+
+  final _GoogleMapTestState mapState;
+  final Completer<GoogleMapController> _controllerCompleter;
+
+  @override
+  _GoogleMapTestState createState() => mapState;
+}
+
+class _GoogleMapTestState extends State<GoogleMapTest> {
+  bool _compassEnabled = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Directionality(
+      textDirection: TextDirection.ltr,
+      child: GoogleMap(
+        initialCameraPosition: _kInitialCameraPosition,
+        compassEnabled: _compassEnabled,
+        onMapCreated: (GoogleMapController controller) {
+          widget._controllerCompleter.complete(controller);
+        },
+      ),
+    );
+  }
+
+  void toggleCompass() {
+    setState(() {
+      _compassEnabled = !_compassEnabled;
+    });
+  }
+}
+
+void main() {
+  final Completer<String> allTestsCompleter = Completer<String>();
+  enableFlutterDriverExtension(handler: (_) => allTestsCompleter.future);
+
+  tearDownAll(() => allTestsCompleter.complete(null));
+
+  _GoogleMapTestState mapTestState;
+  GoogleMapController controller;
+  GoogleMapTestController testController;
+
+  setUp(() async {
+    mapTestState = _GoogleMapTestState();
+    final Completer<GoogleMapController> controllerCompleter =
+        Completer<GoogleMapController>();
+    runApp(GoogleMapTest(mapTestState, controllerCompleter));
+    controller = await controllerCompleter.future;
+    testController = GoogleMapTestController(controller.getMethodChannel());
+  });
+
+  test('testCompassToggle', () async {
+    bool compassEnabled;
+
+    compassEnabled = await testController.isCompassEnabled();
+    expect(compassEnabled, false);
+
+    mapTestState.toggleCompass();
```

Comment:
Looking just at this test case (e.g if I were coming here to debug a test failure), it's kind of hard to understand what's going on.

I tend to favor keeping test cases more verbose, and having repetition across test cases such that when you read the test case you know what it is doing without having to learn the common facilities shared across the test cases in this file.

In this case I'd prefer to see something like:
```dart
pumpWidget(Directionality(
      textDirection: TextDirection.ltr,
      child: GoogleMap(
        initialCameraPosition: _kInitialCameraPosition,
        compassEnabled: false,
        onMapCreated: (GoogleMapController controller) {
          widget._controllerCompleter.complete(controller);
        },
      ),
    ));
 
 compassEnabled = await testController.isCompassEnabled();
    expect(compassEnabled, false);

pumpWidget(Directionality(
      textDirection: TextDirection.ltr,
      child: GoogleMap(
        initialCameraPosition: _kInitialCameraPosition,
        compassEnabled: true,
        onMapCreated: (GoogleMapController controller) {
          widget._controllerCompleter.complete(controller);
        },
      ),
    ));

compassEnabled = await testController.isCompassEnabled();
    expect(compassEnabled, true);
```

More verbose, but it's clear what the test is doing (also I believe this is the typical style of widget tests in the framework)

Reply:
Created `test_widgets.dart` that has `pumpWidget`, we can potentially move it to a shared location, but fits nicely into the framework pattern.